packages/verification-core/src/pdf/PdfService.js

```diff
@@ -0,0 +1,53 @@
+import Bowser from 'bowser'
+import PdfWasm from './pdf.wasm.init'
+import PdfWorker from 'web-worker:./pdf.worker'
+
+export default class PdfService {
+    constructor(pdfWasmUrl) {
+        this.pdfWasmUrl = new URL(pdfWasmUrl, location.origin)
+
+        const browser = Bowser.getParser(window.navigator.userAgent)
+        this.isNonChromiumEdge = browser.satisfies({ edge: '<=18' })
+
+        if (this.isNonChromiumEdge === true) {
+            PdfWasm.init(this.pdfWasmUrl.href)
+        } else {
+            this.worker = new PdfWorker()
+            this.worker.postMessage({
+                cmd: 'init',
+                pdfWasmUrl: this.pdfWasmUrl.href
+            })
+        }
+    }
+
+    /**
+     * Extract encryption keys from pdf
+     * @param pdfBytes
+     * @returns {Promise<Object>}
+     */
+    extractEncryptionKeys(pdfBytes) {
+        return new Promise((resolve, reject) => {
+            if (this.isNonChromiumEdge === true) {
```

Comment:
See comment above.  Maybe `this.isChromiumEdge === false`.

Reply:
This doesn't work, because it's also false for all other browsers.